### ISSUE TITLE
Allow guest to view all post

### ISSUE THREAD

**Issue by kavierkoo** [cnvs/canvas#473]
Hi,

Apologize if newbie question asked, newbie hobbyist trying to develop something here.

I am thinking to make a personal blog as portfolio and came across Canvas, it is awesome, clean and it works.

bumped into few errors like defaultStringLength and MariaDB not support JSON, managed to solve it.

and I am lost as  my objective is to allow visitor to see all published posts without require to Login but not able edit anything.

I can see middleware in vendor\cnvs\canvas\config\canvas.php
also routes mention that initial xx.xx.xx.com/canvas using StatsController 
but I have no clue how to modify to achieve my objective.

Could shed some light on how could I achieve that?

Thanks in advance!

**Comment by austintoddj** [comment]
Glad you like it!

As far as your objective is concerned, are you saying that you'd like visitors to be able to access the /canvas routes but not able to create/edit/delete things?

The middleware you're seeing is simply verifying that authenticated users are accessing Canvas, since the user id is stored in the posts table to notate authorship. 

If you wouldn't mind re-wording your question, perhaps I could be of more help.

**Comment by kavierkoo** [comment]
Hi Austin,
Thanks for the patients and my apology for the confusion, Let me rephrase my words

The goal is just like blogs on blogspot / wordpress where:
- **Author** require to access /login and login to publish / create / modify / delete posts. 
- **Visitor** can access 127.0.0.1:8000/ and see all the posts in canvas with no login required

**Conclusion**:
2 types of user:
- Visitor, No login required, No permission to do anything, Only view all the posts
- Author, login required, Can do anything (actually there will be only 1 author which is me)

I have no clue on how to achieve to only use Middleware when someone access 127.0.0.1:8000/login
but all Posts are visible to all visitors with no login required

I only manage to do some changes on the routes and views to fit my personal requirement. 

So would be very appreciate if you shed some light on how to do so.

Thanks
Regards

**Comment by austintoddj** [comment]
Ah, I think I see where you're going. Correct me if I'm wrong..

Canvas has a built-in Artisan command to generate routes, views, and a controller which makes the posts(published), tags, and topics visible to non-authenticated users. In simpler terms, it generates a frontend.

Check the documentation [section here](https://github.com/cnvs/canvas#configuration) and simply run `php artisan canvas:setup` and you'll be able to visit `http://your-app.test/blog` to see the content you've created with Canvas.

Does that answer your question?

**Comment by kavierkoo** [comment]
Thanks for the prompt response, let me check and test it out first!
Appreciate for the advise!!
Come back to you if I am still stuck!! 

Thanks Austin!!

**Comment by kavierkoo** [comment]
Hi Austin,

It works as you describe!! 
Apologize for overlooked the `php artisan canvas:setup` in documentation!

 The blog looks awesome!!

Thanks!! Kudos to yall!!

**Comment by noahdigesu** [comment]
Hi !

I would like to achieve the exact same result as @kavierkoo, however the `php artisan canvas:setup` command is not available anymore.
How should I go about this ?

Thank you in advance :)

**Comment by austintoddj** [comment]
@noahdigesu The command you're looking for is `install`. See it [here in the documentation](https://github.com/austintoddj/canvas#installation).

**Comment by noahdigesu** [comment]
Nice ! Thanks ! :D
I feel like the README might not be super clear about this - I thought `canvas:ui` was an alternative to the "default" look of canvas.
Maybe mention "Want to enable guest users to access your articles ?" or something similar to make it more explicit ?

**Comment by austintoddj** [comment]
Understood. Thanks for brining up that it was difficult reading between the lines on that. I'll see what it looks like updating that in the next release.
